### ISSUE TITLE
Parenthesis in trailing closure rule

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#885]
When using trailing closures, one can add `()` after the method call, but that should be discouraged.

```swift
let x = [1, 2]

// shouldn't trigger
_ = x.map { $0 + 1 }

// should trigger
_ = x.map() { $0 + 1 }
```

Note that there are cases when parenthesis are allowed (when there're more than 1 parameter):

```swift
// shouldn't trigger
_ = x.reduce(0) { $0 + $1 }
```

**Comment by jpsim** [comment]
`EmptyParenthesesWithTrailingClosureRule`. A bit verbose, but descriptive.
